### PR TITLE
Fix : Revamp yocto build for any platform

### PR DESCRIPTION
**Pull request by nbmaiti**
Remove file based build, 
1) Use safestring build based on platform.
2) use git client-sdk repository for source build dynamically. Remove zip based src file  ,
3) Can be used for target ARM/x86 platforms for yocto build

Signed-off-by: Nabendu Maiti <nabendu.bikash.maiti@intel.com>

### CODE REVIEW

File: utils/meta-intel-sdo/recipes-connectivity/ccodesdk/ccodesdk.bb [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#sdo yocto project
+
+DESCRIPTION = "Recipe for sdo (c-code-sdk) on linux"
```
Comment (AdithyaBaglody) [inline utils/meta-intel-sdo/recipes-connectivity/ccodesdk/ccodesdk.bb:3]: need to rename the files are clientsdk.bb and need to remove the ref to c-code-sdk.

File: utils/meta-intel-sdo/recipes-connectivity/c-code-sdk/c-code-sdk.bb [hunk 2]
```diff
@@ -1,51 +0,0 @@
...
-SRC_URI = "file://sdo.tar.gz"
-
-
-do_configure(){
-}
-
-do_compile(){
-CUR_DIR=$(pwd)
-cd "${WORKDIR}/sdo"
-export CFLAGS="-D_IPP_v50_ -D_IPP_DEBUG -D_DISABLE_ALG_MD5_ -mfloat-abi=hard -D_DISABLE_ALG_SM3_ -Wstrict-aliasing -g -D_FORTIFY_SOURCE=2 HTTPPROXY=false"
```
Comment (AdithyaBaglody) [inline utils/meta-intel-sdo/recipes-connectivity/c-code-sdk/c-code-sdk.bb:30]: this is needed for ARM build. how is this handled in the new .bb files??

**Review by AdithyaBaglody — CHANGES_REQUESTED** [review]

File: utils/meta-intel-sdo/recipes-connectivity/c-code-sdk/c-code-sdk.bb (hunk 2 above)
Comment (nbmaiti) [reply utils/meta-intel-sdo/recipes-connectivity/c-code-sdk/c-code-sdk.bb:30]: by cmake.

File: utils/meta-intel-sdo/recipes-connectivity/ccodesdk/ccodesdk.bb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+cd ${SAFESTRING_ROOT}
+rm -rf makefile
+sed -i '/mmitigate-rop/d' ./CMakeLists.txt
+cmake .
+make
+cp libsafestring_static.a libsafestring.a
+
+cd ${S}
+cmake -DPK_ENC=ecdsa -DDA=ecdsa256 -DMANUFACTURER_TOOLKIT=true -DKEX=ecdh . ;
+CFLAGS="${CFLAGS}" make -j$(nproc)
```
[the remark this answers is not in the document]
Comment (nbmaiti) [reply utils/meta-intel-sdo/recipes-connectivity/ccodesdk/ccodesdk.bb:47]: cflags are not used, its mistakenly added

File: utils/meta-intel-sdo/recipes-connectivity/ccodesdk/ccodesdk.bb (hunk 1 above)
Comment (nbmaiti) [reply utils/meta-intel-sdo/recipes-connectivity/ccodesdk/ccodesdk.bb]: done

File: utils/meta-intel-sdo/recipes-connectivity/c-code-sdk/c-code-sdk.bb (hunk 2 above)
Comment (AdithyaBaglody) [reply utils/meta-intel-sdo/recipes-connectivity/c-code-sdk/c-code-sdk.bb:30]: currently its not in cmake.

File: utils/meta-intel-sdo/recipes-connectivity/clientsdk-mbedtls/clientsdk-mbedtls.bb [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+export SAFESTRING_ROOT=${CUR_DIR}/../safestringlib
+cd ${SAFESTRING_ROOT}
+rm -rf makefile
+sed -i '/mmitigate-rop/d' ./CMakeLists.txt
+cmake .
+make
+cp libsafestring_static.a libsafestring.a
+
+cd ${S}
+cmake -DTLS=mbedtls -DPK_ENC=ecdsa -DDA=ecdsa256 -DMANUFACTURER_TOOLKIT=true -DKEX=ecdh . ;
```
Comment (AdithyaBaglody) [inline utils/meta-intel-sdo/recipes-connectivity/clientsdk-mbedtls/clientsdk-mbedtls.bb:53]: -DMANUFACTURER_TOOLKIT=true 
can be removed

**Review by AdithyaBaglody — APPROVED** [review]
LGTM

File: utils/meta-intel-sdo/recipes-connectivity/clientsdk-mbedtls/clientsdk-mbedtls.bb [hunk 5]
```diff
@@ -0,0 +1,79 @@
+#sdo yocto project
+
+DESCRIPTION="Recipe for sdo (client-sdk) on linux with mbedtls"
```
Comment (trbehera) [inline utils/meta-intel-sdo/recipes-connectivity/clientsdk-mbedtls/clientsdk-mbedtls.bb:3]: nitpick:
sdo => SDO

File: utils/meta-intel-sdo/recipes-connectivity/clientsdk/clientsdk.bb [hunk 6]
```diff
@@ -0,0 +1,75 @@
+#sdo yocto project
+
+DESCRIPTION = "Recipe for sdo (client-sdk) on linux"
```
Comment (trbehera) [inline utils/meta-intel-sdo/recipes-connectivity/clientsdk/clientsdk.bb:3]: nitpick: sdo => SDO

**Review by trbehera — APPROVED** [review]
There are a couple of minor nitpicks, fix if reposting.

File: utils/meta-intel-sdo/recipes-connectivity/clientsdk-mbedtls/clientsdk-mbedtls.bb (hunk 5 above)
Comment (nbmaiti) [reply utils/meta-intel-sdo/recipes-connectivity/clientsdk-mbedtls/clientsdk-mbedtls.bb]: done

File: utils/meta-intel-sdo/recipes-connectivity/clientsdk/clientsdk.bb (hunk 6 above)
Comment (nbmaiti) [reply utils/meta-intel-sdo/recipes-connectivity/clientsdk/clientsdk.bb]: done

**Review by AdithyaBaglody — APPROVED** [review]

**Pull request merged**